File: app/src/main/java/com/alex/morgan/bearlist/list/BearListComponent.java
package com.alex.morgan.bearlist.list;

import com.alex.morgan.bearlist.app.AppComponent;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(dependencies = AppComponent.class, modules = BearListModule.class)
public interface BearListComponent {
    void inject(BearListActivity activity);
}

File: app/src/main/java/com/alex/morgan/bearlist/list/BearListComponent.kt
package com.alex.morgan.bearlist.list

import com.alex.morgan.bearlist.BearListFragment
import com.alex.morgan.bearlist.app.ActivityScope
import com.alex.morgan.bearlist.app.AppComponent

import javax.inject.Singleton

import dagger.Component

@ActivityScope
@Component(dependencies = [(AppComponent::class)], modules = [(BearListModule::class)])
interface BearListComponent {
    fun inject(activity: BearListActivity)

    fun inject(fragment: BearListFragment)
}
